ip-whitelist/cmd/resource/resource.go

```diff
@@ -0,0 +1,218 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/encoding"
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/rs/xid"
+)
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey.Value(), *currentModel.ApiKeys.PrivateKey.Value())
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	projectID := *currentModel.ProjectId.Value()
+	request := getProjectIPWhitelistRequest(currentModel)
+
+	_, _, err = client.ProjectIPWhitelist.Create(context.Background(), projectID, request)
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	guid := xid.New()
+
+	currentModel.Id = encoding.NewString(guid.String())
```

Comment:
Should we not rely on an id returned by the call made to `client.ProjectIPWhitelist.Create` like we do in https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/7/files#diff-7da7a1a385b2ed8cb2a84a39be33b28c or is generating our own id better because we need to keep track of the whitelist seperately from Atlas?

Reply:
IP whitelist does not has an ID https://github.com/mongodb/go-client-mongodb-atlas/blob/master/mongodbatlas/project_ip_whitelist.go#L33

Reply:
I was tracking them with the ID, but I can track them with prev and current Model inside the function, since I am already deleting and recreating everything I don't need to track them with the ID I was creating. In order to avoid a super large ID once multiple whitelist entries are added I will just create one for this.